mpl/request.hpp

```diff
@@ -193,7 +226,8 @@ namespace mpl {
       /// Cancels a pending request in the pool.
       /// \param i index of the request for which shall be cancelled
       void cancel(size_type i) {
-        MPI_Cancel(&requests_[i]);
+        if (requests_[i] != MPI_REQUEST_NULL)
+          MPI_Cancel(&requests_[i]);
```

Comment:
Actually I'm not quite sure about this. I added this check since I notice that it is not possible for user to (incorrectly?) push an already moved request into the pool. So when I noticed that `base_request::cancel` has this check, I added also here. But if there's an assumption that user will never push a null request into the pool, this check can be removed for consistency.

Reply:
I need to double-check this. But I think the check against `MPI_REQUEST_NULL` *is* actually needed here. Thanks for spotting this.